base/strings/search.jl

```diff
@@ -23,176 +44,190 @@ julia> search("Hello to the world", "z")
 
 julia> search("JuliaLang","Julia")
 1:5
+
+julia> search(Int8[1,2,3], Int8[2,3])
+2:3
+
+julia> search("Hello to the world", 'z')
+0
+
+julia> search("Hello to the world", ['e', 'l'])
+2
+
+julia> search("", ['e', 'l'])
+0
 ```
 """
 function search(s::AbstractString, c::Chars, i::Integer)
-    if isempty(c)
-        return 1 <= i <= nextind(s,endof(s)) ? i :
-               throw(BoundsError(s, i))
-    end
-    if i < 1 || i > nextind(s,endof(s))
-        throw(BoundsError(s, i))
-    end
-    while !done(s,i)
-        d, j = next(s,i)
+    isvalid(s, i) || throw(ArgumentError("index $i is invalid"))
+    isempty(c) && return 0
+    while !done(s, i)
+        d, j = next(s, i)
         if d in c
             return i
         end
         i = j
     end
     return 0
 end
-search(s::AbstractString, c::Chars) = search(s,c,start(s))
 
-in(c::Char, s::AbstractString) = (search(s,c)!=0)
+search(s::AbstractString, c::Chars) = s == "" ? 0 : search(s, c, start(s))
+
+in(c::Char, s::AbstractString) = search(s,c) != 0
```

Comment:
Plase don't change unrelated lines. Same below. This makes it really painful to review.